spring-boot-tools/spring-boot-maven-plugin/src/main/java/org/springframework/boot/maven/RunArguments.java

```diff
@@ -31,14 +32,17 @@
 
 	private static final String[] NO_ARGS = {};
 
-	private final LinkedList<String> args;
+	private final LinkedList<String> args = new LinkedList<>();
 
 	RunArguments(String arguments) {
 		this(parseArgs(arguments));
 	}
 
 	RunArguments(String[] args) {
-		this.args = new LinkedList<>(Arrays.asList(args));
+		if (args != null) {
+			this.args.addAll(Arrays.stream(args).filter(s -> s != null && !"".equals(s))
```

Comment:
We could probably replace the filter with `hasLength` from `org.springframework.util.StringUtils`. A `forEach` might also be a bit more concise than the collector:

```java
Arrays.stream(args).filter(StringUtils::hasLength).forEach(this.args::add);
```